RELEASES.md

```diff
@@ -1,3 +1,116 @@
+- [Update `indirect_structural_match` and `pointer_structural_match` lints to match RFC](https://github.com/rust-lang/rust/pull/120423/)
+- [Make non-`PartialEq`-typed consts as patterns a hard error](https://github.com/rust-lang/rust/pull/120805/)
+- [align_offset, align_to: no longer allow implementations to spuriously fail to align](https://github.com/rust-lang/rust/pull/121201/)
+- [Split refining_impl_trait lint into `_reachable`, `_internal` variants](https://github.com/rust-lang/rust/pull/121720/)
+- [Make `ConstPropLint` lint run on promoteds](https://github.com/rust-lang/rust/pull/119432/)
+
+<a id="1.78.0-Compiler"></a>
+
+Compiler
+--------
+
+- [Made `INVALID_DOC_ATTRIBUTES` lint deny by default](https://github.com/rust-lang/rust/pull/111505/)
+- [Extend `invalid_reference_casting` to include references casting to bigger memory layout](https://github.com/rust-lang/rust/pull/118983/)
+- [Increase accuracy of redundant `use` checking](https://github.com/rust-lang/rust/pull/117772/)
+- [Suggest moving definition if non-found macro_rules! is defined later](https://github.com/rust-lang/rust/pull/121130/)
+- [Lower transmutes from int to pointer type as gep on null](https://github.com/rust-lang/rust/pull/121282/)
+- [Update to LLVM 18](https://github.com/rust-lang/rust/pull/120055/)
+
+Target changes:
+
+- [Increase minimum to Windows 10 for tier 2 platform support](https://github.com/rust-lang/rust/pull/115141/)
+  - [Enables CMPXCHG16B, SSE3, SAHF/LAHF and 128-bit Atomics in tier 1 Windows](https://github.com/rust-lang/rust/pull/120820/)
+- [Add `wasm32-wasip1` tier 2 (without host tools) target](https://github.com/rust-lang/rust/pull/120468/)
+- [Add `arm64ec-pc-windows-msvc` tier 3 target](https://github.com/rust-lang/rust/pull/119199/)
+- [Add `loongarch64-unknown-linux-musl` tier 3 target](https://github.com/rust-lang/rust/pull/121832/)
+- [Add `armv8r-none-eabihf` tier 3 target for the Cortex-R52](https://github.com/rust-lang/rust/pull/110482/)
+- [Rename `wasm32-wasi-preview1-threads` to `wasm32-wasip1-threads`](https://github.com/rust-lang/rust/pull/122170/)
+- [Add a new `wasm32-wasi-preview2` target](https://github.com/rust-lang/rust/pull/119616/)
+
+- [coverage: Split out counter increment sites from BCB node/edge counters](https://github.com/rust-lang/rust/pull/120564/)
+- [coverage: Improve handling of function/closure spans](https://github.com/rust-lang/rust/pull/120569/)
+- [Only generate a ptrtoint in AtomicPtr codegen when absolutely necessary](https://github.com/rust-lang/rust/pull/122220/)
+
+<a id="1.78.0-Libraries"></a>
+
+Libraries
+---------
+- [Optimize away poison guards when std is built with panic=abort](https://github.com/rust-lang/rust/pull/100603/)
+- [PartialEq, PartialOrd: update and synchronize handling of transitive chains](https://github.com/rust-lang/rust/pull/115386/)
+- [Toggle assert_unsafe_precondition in codegen instead of expansion](https://github.com/rust-lang/rust/pull/120594/)
```

Comment:
I think `assert_unsafe_precondition!` is an internal macro, should this therefor be put in the "Internal" section?

Reply:
My suspicion is that this should be re-written to point out that users will now see the effects of unsafe precondition checking in many standard library functions (cc https://github.com/rust-lang/rust/issues/123285).

But this is a user-visible change, so maybe it moves to compatibility notes, but it's very likely to stay outside of internal.